lib/application.ts

```diff
@@ -930,19 +930,25 @@ export class SNApplication {
   }
 
   /**
-   * Creates a JSON string representing the backup format of all items, or just subItems
-   * if supplied.
+   * Creates a JSON-stringifiable backup object of all items.
    */
   public async createBackupFile(
-    subItems?: SNItem[],
-    intent?: EncryptionIntent,
-    returnIfEmpty = false
-  ) {
-    return this.protocolService!.createBackupFile(
-      subItems,
-      intent,
-      returnIfEmpty
-    );
+    intent: EncryptionIntent
+  ): Promise<BackupFile | undefined> {
+    const items = this.itemManager.items;
+
+    if (intent === EncryptionIntent.FileDecrypted) {
+      if (items.some(item => item.protected) && this.hasPasscode()) {
+        const passcode = this.challengeService.promptForPasscode(
```

Comment:
Shouldn't this be awaited? And how is passcode being used? The return type for promptForPasscode isn't even a passcode, it's an object containing the passcode. Sounds like this functionality should be covered by a test.

Reply:
Oh I see, promptForPasscode is refactored in this PR. But still needs to be awaited.